matchzoo/datapack.py

```diff
@@ -53,18 +53,36 @@ def __init__(self,
         :param left: Store the content or features for id_left.
         :param right: Store the content or features for
             id_right.
-        :param context: Hyper-parameter fitted during
-            pre-processing stage.
         """
         self._relation = relation
         self._left = left
         self._right = right
-        self._context = context
+
+    @property
+    def stage(self):
+        if 'label' in self._relation.columns:
+            return 'train'
+        else:
+            return 'predict'
 
     def __len__(self) -> int:
         """Get numer of rows in the class:`DataPack` object."""
         return self._relation.shape[0]
 
+    def __getitem__(self, index):
+        if isinstance(index, int):
+            index = [index]
+        elif isinstance(index, slice):
+            index = list(range(*index.indices(len(self))))
+        left_ids = self._relation['id_left'][index]
+        right_ids = self._relation['id_right'][index]
+        left_df = self._left.loc[left_ids].reset_index().set_index([index])
+        right_df = self._right.loc[right_ids].reset_index().set_index([index])
+        ret = left_df.join(right_df)
```

Comment:
what is `ret`? return?

Reply:
Yes. It's a commonly used abbreviation for return value.